R/scregclust.R

```diff
@@ -1367,6 +1393,63 @@ scregclust <- function(expression,
           }
         }
 
+      ###############################################
+      ## START Step 1.5: Determining target genes  ##
+      ###############################################
+
+      # With quick_mode = True a subset of target genes are selected from cycle two and onward.
+      # For each of the targets inside the noise module we determine the biggest (in absolute value)
+      # correlation to some active regulator. Then we take the top "percent" of the noise targets
+      # based on this value, together with the targets belonging to some active module, and this makes
+      # up our collection of target genes for which we calculated NNLS and do re-allocation
+
+
+      
+      if (cycle != 1 && sum(models) != 0 && quick_mode && percent != 1) {
+        # Get current active regulators and targets for subsetting the correlation matrix
+        active_regulators <- which(apply(models, 1, sum) >= 1)
+        active_targets <- which(idx != -1)
+
+        # If we are in the last cycle we only need to estimate the active targets
+        if (last_cycle){
+          filtered_targets <- active_targets
+        }
+        else{
+        # Use as.matrix to be able to use apply even when there is a single regulator
+        abs_corr_matrix_temp <- as.matrix(abs_corr_matrix[-active_targets, active_regulators])
+
+        # Create a vector of maximal absolute correlation
+        max_abs_correlation <- apply(abs_corr_matrix_temp, 1, function(row) max(row, na.rm = TRUE))
```

Comment:
is this necessary? Never saw NAs in the correlation matrix

Reply:
You are correct, not necessary, removed.